Allow specifying multiple Jenkins URLs

nre-ableton opened the pull request:
Ping @AbletonDevTools/gotham-city

nre-ableton committed:
Move Jenkins object declaration closer to usage

This will force a new connection to the master on each loop iteration,
but it will also allow us to support multiple Jenkins masters. That
feature will be added in a future commit.

nre-ableton committed:
Extract get_node_infos function from main loop

rco-ableton reviewed (approved):
LGTM 👍

ala-ableton commented on Jenkinsfile:
```diff
@@ -49,7 +49,7 @@ devToolsProject.run(
       '-v jenkins-nodes:/jenkins_nodes',
       '--target-port 9100 --target-port 9323' +
-        " ${env.JENKINS_URL} /jenkins_nodes/output.json",
+        " --url ${env.JENKINS_URL} /jenkins_nodes/output.json",
```
This change belongs to _Move Jenkins URL to required --url argument_ instead of to _Allow specifying multiple Jenkins URLs_.

nre-ableton replied:
Right you are!

nre-ableton committed:
Allow specifying multiple Jenkins URLs

ala-ableton commented on jenkins_node_scanner.py:
```diff
@@ -224,6 +216,14 @@ def main():
     while True:
         with ignore_exceptions_except_exit():
             with GLOBAL_EXCEPTIONS.count_exceptions():
+                logging.info('Connecting to Jenkins master at %s', args.url)
```
How much spam is this producing then?

nre-ableton replied:
Not much, but I guess it should be a debug line.

mst-ableton commented on jenkins_node_scanner.py:
```diff
@@ -224,30 +243,22 @@ def main():
+                        timeout=args.timeout,
+                    )
+
+                    node_infos.append(
```
Shouldn't this be `node_infos.extend(...)`?

nre-ableton replied:
Good catch, fixed in 475a696f6cdb387c69ad2323e1c947dcb85464f9

ala-ableton replied:
Then I guess `get_node_info` could have a better name.

mst-ableton reviewed (changes requested):
lgtm except for one change

ala-ableton commented on jenkins_node_scanner.py:
```diff
@@ -158,6 +158,27 @@ def get_node_ip(master, node):
     return description.split('from ')[1].split(' :')[0]
 
 
+def get_node_info(master, target_ports, exclude_regex):
```
This function return several items in a list, so it should be named `get_node_infos` or `get_nodes_info`.

nre-ableton replied:
Agreed. I prefer the first option. Fixed in 6ebf4502ce8450894f658b07efd5d04cbbe8d1fa.

ala-ableton commented:
@nre-ableton please rebase/squash and I'll do another review, thanks!

nre-ableton commented:
@ala-ableton done!

ala-ableton commented on jenkins_node_scanner.py:
```diff
@@ -224,6 +216,14 @@ def main():
     while True:
         with ignore_exceptions_except_exit():
             with GLOBAL_EXCEPTIONS.count_exceptions():
+                logging.info('Connecting to Jenkins master at %s', args.url)
```
I think the `logging.info` -> `logging.debug` change belongs to _Move Jenkins object declaration closer to usage_.

ala-ableton commented on jenkins_node_scanner.py:
```diff
@@ -224,28 +245,7 @@ def main():
-                                if is_port_open(ip_addr, port)],
-                } for node, ip_addr in node_ips]
+                node_info = get_node_infos(master, args.target_ports, exclude_regex)
 
```
Please rename `node_info` to `node_infos` to match `get_node_infos`.

nre-ableton committed:
Move Jenkins URL to required --url argument

This will ultimately allow for specifying multiple Jenkins masters,
but also makes the CLI syntax a bit cleaner by reducing the number of
positional arguments.

ala-ableton reviewed (approved):
Very nice!
LGTM :+1:
